src/main/java/com/hollingsworth/arsnouveau/common/entity/AmethystGolem.java

```diff
@@ -112,6 +121,19 @@ public MinecoloniesAdvancedPathNavigate getNavigation() {
         return pathNavigate;
     }
 
+    @Override
+    public float getWalkTargetValue(BlockPos pos, LevelReader level) {
```

Comment:
We should just make a new goal and replace `WaterAvoidingRandomStrollGoal` with a goal that only picks positions near the home position.

Reply:
ill drop this change for now